src/main/scala/gestalt/core/Trees.scala

```diff
@@ -131,11 +131,13 @@ trait Trees extends Params with TypeParams with
 
   def NewInstance: NewInstanceImpl
   trait NewInstanceImpl {
-    def apply(qual: Option[Tree], name: String, targs: List[TypeTree], argss: List[List[TermTree]]): TermTree
-    def apply(tpe: TypeTree, argss: List[List[TermTree]]): TermTree = {
-      val Trees.this.PathType(qual, name, targs) = tpe
-      apply(qual, name, targs, argss)
-    }
+    def apply(qual: Option[Tree],
+              name: String,
+              targs: List[TypeTree],
+              argss: List[List[TermTree]])(implicit unsafe: Unsafe): TermTree
```

Comment:
Thanks a lot for this PR @valdisxp1 . It prompts me to think more about the design.

The most important motivation for making `qual`, `name`, `targs` explicit in the constructor is to make the extractor more friendly. The consideration for semantic difference is secondary.

While it's possible to make this constructor `private[gestalt]` and define the following helpers in `api`:

```Scala
def NewInstance(name: String, targs: List[TypeTree], argss: ...)(implicit unsafe: Unsafe)
def NewInstance(qual: Tree, name: String, targs: List[TypeTree], argss: ...)
```

But I tend to think the above is over-design. I'd like to abandon the original design due following reasons:

- inspecting syntactic structure of New expression is useless, only crappy macros do that
- even if we merge `qual`, `name` and `targs`, it's still not difficult to address potential semantic different in compiler implementation.
- following code is valid in Scala, which implies the semantic consideration is not well founded.

```Scala
type A = Some[Int]
new A(3)
```

Due to the consideration above, I'd like to suggest to change the signature of the `apply` to the follows:

```Scala
def apply(typ: TypeTree, argss: List[List[TermTree]]): TermTree
```

Another change we can do is to add `package macros` to all macro definition files.

With these two changes, we can avoid the unsafe import at https://github.com/liufengyun/gestalt/pull/89/files#diff-f6a7574868d1bbd54893d1ef5b7e8aa1R32 by using `new _root_.macros.Optional`